AxonDeepSeg/train_network.py

```diff
@@ -54,588 +58,178 @@ def train_model(path_trainingset, path_model, config, path_model_init=None,
-                #### d) Saving the model as a checkpoint, the metrics in a pickle file and update the file report.txt
-                ### ----------------------------------------------------------------------------------------------- ###
-
-                # Saving the model if it's the best one. We only do this check at the end of an epoch
-                if epoch == 1: # First epoch is 1, not 0
-                    acc_current_best = acc
-                    loss_current_best = loss
-
-                # At each check frequency defined we look if the moving average is better
-                elif ((epoch >= max(save_best_moving_avg_epoch_freq, save_best_moving_avg_window)) and (epoch%save_best_moving_avg_epoch_freq == 0)):
-                    acc_moving_avg = np.mean(Accuracy[-save_best_moving_avg_window:])
-                    loss_moving_avg = np.mean(Loss[-save_best_moving_avg_window:])
-                    if acc_moving_avg > acc_current_best:
-
-                        save_path = saver.save(session, str(path_model / "best_acc_model.ckpt"))
-                        acc_current_best = acc_moving_avg
-                        print(("Best accuracy model saved in file: %s" % save_path))
-
-                        # Saving the evolution in a pickle file
-                        evolution = {'loss': np.mean(Loss[-10:]),
-                                     'steps': epoch,
-                                     'accuracy': np.mean(Accuracy[-10:])}
-
-                        with open(path_model / 'best_acc_stats.pkl', 'wb') as handle:
-                            pickle.dump(evolution, handle)
-
-                    if loss_moving_avg < loss_current_best:
-
-                        save_path = saver.save(session, str(path_model / "best_loss_model.ckpt"))
-                        loss_current_best = loss_moving_avg
-                        print(("Best loss model saved in file: %s" % save_path))
-
-                        # Saving the evolution in a pickle file
-                        evolution = {'loss': np.mean(Loss[-10:]),
-                                     'steps': epoch,
-                                     'accuracy': np.mean(Accuracy[-10:])}
-
-                        with open(path_model / 'best_loss_stats.pkl', 'wb') as handle:
-                            pickle.dump(evolution, handle)
-
-                # Moreover at a frequency we save the model regardless of the performance
-                if epoch % save_last_epoch_freq == 0:
-
-                    evolution = {'loss': Loss, 'steps': Epoch, 'accuracy': Accuracy}
-                    with open(path_model / 'evolution.pkl', 'wb') as handle:
-                        pickle.dump(evolution, handle)
-                    save_path = saver.save(session, str(path_model / "model.ckpt"))
+    # Data dictionary to feed into image generator
+    data_gen_args = dict(horizontal_flip=True,
```

Comment:
From the look of the commented-out arguments, I think the data augmentation implementation is incomplete?